### PR TITLE
Fixes Construction Progress Bars

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Finally fixes there not being progress bars for most forms of stack-based construction

:cl: Fox McCloud
fix: Fixes the lack of progress bars for more stack-based construction
/:cl:

### CODE REVIEW

**Comment by taukausanake** [comment]
Progress bars are still a bit slow due to client and server mismatches, correct? Just want to make sure as it is related

**Comment by Fox-McCloud** [comment]
That's 100% a BYOND thing. Recommend using a different version. 

I'm running 510.1336 and have no issues with progress bars unless there's server lag or I'm lagging.

File: code/game/objects/items/stacks/rods.dm [hunk 1]
```diff
@@ -80,7 +80,7 @@
 			return
 		to_chat(usr, "\blue Assembling grille...")
 
-		if (!do_after(usr, 10, target = src))
+		if (!do_after(usr, 10, target = user))
```
Comment (monster860) [inline code/game/objects/items/stacks/rods.dm]: Change user to usr

`if (!do_after(usr, 10, target = usr))`

**Comment by QQCucumber** [comment]
Relating to the server mismatches on the progress bar stuff: #4579 

1540 completely broke smooth progress bars, 1541 fixed them but introduced a compatibility problem. Both the server and the client need to have 1536-1539 or 1541-1545 for it to work properly.

Hopefully the server updates to stable sooner rather than later.

File: code/game/objects/items/stacks/rods.dm (hunk 1 above)
Comment (tigercat2000) [inline code/game/objects/items/stacks/rods.dm]: uh no, change `usr` to `user`

Comment (monster860) [inline code/game/objects/items/stacks/rods.dm]: Do that for the whole proc though.

**Pull request merged**